include/mbedtls/ssl_internal.h

```diff
@@ -437,7 +437,7 @@ struct mbedtls_ssl_handshake_params
 #if defined(MBEDTLS_DHM_C)
     mbedtls_dhm_context dhm_ctx;                /*!<  DHM key exchange        */
 #endif
-#if defined(MBEDTLS_ECDH_C)
+#if defined(MBEDTLS_ECDH_C) || defined(PSA_WANT_ALG_ECDH)
     mbedtls_ecdh_context ecdh_ctx;              /*!<  ECDH key exchange       */
```

Comment:
I still don't understand why a guard should change here. Why would `ecdh_ctx` be needed if anything other than built-in ECDH is enabled?

Reply:
@gilles-peskine-arm for ssl.h and ssl_internal.h if MBEDTLS_ECDH_C is not defined but PSA_WANT_ALG_ECDH is defined with accelerators the build will not compile if MBEDTLS_ECDSA_C is also defined. What is the appropriate way to resolve that dependency. Should ssl_cli.c and ssl_srv.c be updated to make the use of ecdh_ctx only when ECDH_C is set and not also allow ECDSA_C in those files?

Reply:
Ok, so apparently it's needed for two occurrences (one in `ssl_srv.c`, one in `ssl_cli.c`) of
```
#if defined(MBEDTLS_ECDH_C) || defined(MBEDTLS_ECDSA_C)
            ssl->handshake->ecdh_ctx.point_format = p[0];
#endif
```
I think `|| defined(MBEDTLS_ECDSA_C)` shouldn't be there, but it didn't matter much because there are (currently) no TLS cipher suites that use ECDSA without ECDH. I'm not sure that just removing `|| defined(MBEDTLS_ECDSA_C)` is the right thing so I've filed it as https://github.com/ARMmbed/mbedtls/issues/3891 (also it's demonstrating a test gap).

Since this is complicated due to a preexisting issue, I'm ok with adding `|| defined(MBEDTLS_ECDSA_C)` here as a workaround. Please add a comment referencing that issue.